### PR TITLE
[Gluten-core] Refactor scan validation

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

Refactor the scan validation logic of different backends.

## How was this patch tested?

Testing on CI.

### CODE REVIEW

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala [hunk 1]
```diff
@@ -36,17 +41,55 @@ class VeloxBackend extends Backend {
 
 object VeloxBackendSettings extends BackendSettings {
   override def supportFileFormatRead(format: ReadFileFormat,
-                                     fields: Array[StructField]): Boolean = {
+                                     fields: Array[StructField],
+                                     paths: Seq[String],
+                                     partNonEmpty: Boolean): Boolean = {
+    // Validate partition info.
+    def validatePartInfo: Boolean = {
+      // Fallback to vanilla spark when the input path does not contain the partition info.
```
Comment (jinchengchenghh) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala:49]: This maybe duplicated https://github.com/oap-project/gluten/pull/961/files#diff-74bb54358117b0553772ea92af4b099ccc113ed949aef65729650653f2642847R81

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala [hunk 2]
```diff
@@ -36,17 +41,55 @@ class VeloxBackend extends Backend {
...
+      true
+    }
+
+    // Validate compression codec based on file metadata.
+    def validateFileMetaData: Boolean = {
+      val conf = new Configuration()
+      val parquetReadOptions = HadoopReadOptions.builder(conf).build()
+      val supportedCodec = Seq("none", "uncompressed", "snappy", "gzip", "zstd")
+      for (path <- paths) {
+        val hadoopInputFile = HadoopInputFile.fromPath(new Path(path), conf)
```
Comment (jinchengchenghh) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala:62]: Does S3 file can read metadata by this way?

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala [hunk 3]
```diff
@@ -36,17 +41,55 @@ class VeloxBackend extends Backend {
...
+        val parquetFile = ParquetFileReader.open(hadoopInputFile, parquetReadOptions)
+        try {
+          parquetFile.getFooter.getBlocks.forEach(block => {
+            block.getColumns.forEach(column => {
+              if (!supportedCodec.contains(column.getCodec.name().toLowerCase())) {
```
Comment (jinchengchenghh) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala:67]: This validate may be too exbensive, especially there is many small files which is common in user production environments.
Can we just check the first file or sample to check file? I suppose files with different codec is rarely.

**Comment by FelixYBW** [comment]
@zzcclp looks a little change to CK.

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala (hunk 1 above)
Comment (rui-mo) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala]: Thank you. Fixed.

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala (hunk 3 above)
Comment (rui-mo) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala]: Thanks for this suggestion. Fixed.

File: gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala [hunk 4]
```diff
@@ -84,8 +84,8 @@ trait BasicScanExecTransformer extends TransformSupport {
       return false
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:84]: @zzcclp Looks this validation is also needed for CH backend. Is that correct?

Comment (zzcclp) [reply gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:84]: yes, CH backend needs this validation, the partition values are read from file path by CH backend.

**Comment by zzcclp** [comment]
> @zzcclp looks a little change to CK.

Yeah, I will raise a another pr to fix this issue first.

**Comment by rui-mo** [comment]
@jinchengchenghh Would you like to review again? This PR only conducts a refactor now.

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by jinchengchenghh — APPROVED** [review]

**Pull request merged**